enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/Utils.java

```diff
@@ -128,4 +133,34 @@ static void populateRemoveAndProtectedHeaders(RequestContext requestContext) {
         // Authorization Header should not be included in the throttle publishing event.
         requestContext.getProtectedHeaders().add(authHeaderName);
     }
+
+    public static ResourceConfig buildResource(Operation operation, String resPath, Map<String,
+            List<String>> apiLevelSecurityList) {
+        ResourceConfig resource = new ResourceConfig();
+        resource.setPath(resPath);
+        resource.setMethod(ResourceConfig.HttpMethods.valueOf(operation.getMethod().toUpperCase()));
+        resource.setTier(operation.getTier());
+        resource.setDisableSecurity(operation.getDisableSecurity());
+        Map<String, List<String>> securityMap = new HashMap<>();
+        if (operation.getSecurityList().size() > 0) {
```

Comment:
This logic has been there for REST for sometime now, but are we sure we don't get any NPEs from here? 
Bdw, the correct check should be a null check here. This current `if` statement with size check doesn't really have an effect. (the for loop won't get executed if there are no elements in `operation.getSecurityList()` anyway)

And here also we are directly gettting `securityList.getScopeListMap().entrySet()` without checking whether `securityList.getScopeListMap()` is null.

```suggestion
        if (operation.getSecurityList() != null) {
```

Reply:
Operation is a generated proto message and it looks like, it is not getting null [1].

There is an `else` part there, so we may have to keep the if condition.

[1] https://stackoverflow.com/questions/20891425/can-java-protocol-buffers-ever-return-null-on-the-list-getter-of-a-repeated-fiel#:~:text=No%2C%20there%20is%20no%20case,allow%20you%20to%20set%20null%20.

Reply:
Ack. Then we are good.